src/libostree/ostree-repo-commit.c

```diff
@@ -2160,9 +2177,13 @@ ostree_repo_transaction_set_collection_ref (OstreeRepo                *self,
                                             const OstreeCollectionRef *ref,
                                             const char                *checksum)
 {
-  g_return_if_fail (OSTREE_IS_REPO (self));
-  g_return_if_fail (self->in_transaction == TRUE);
-  g_return_if_fail (ref != NULL);
+  g_assert (self != NULL);
+  g_assert (OSTREE_IS_REPO (self));
+  g_assert (self->in_transaction == TRUE);
+  g_assert (ref != NULL);
+
+  // TODO(lucab): introduce a method with error-returning in order to deprecate
+  // this one, because it can silently fail.
```

Comment:
Is there a reason you didn't want to migrate it to `g_assert`?

Reply:
This one below is not a programming error, but it triggers depending on the given `checksum` input.
So this should be a normal runtime error and signal back the validation failure, but the current signature cannot accommodate that.

Under the current API I don't see a clear winning move out of this problematic situation, so I opted for leaving it untouched.